torch_xla/csrc/ir.cpp

```diff
@@ -102,14 +102,18 @@ std::string Use::ToString() const {
   return ss.str();
 }
 
-const xla::Shape& Value::xla_shape() const { return node->xla_shape(index); }
-
-const xla::Shape& Value::xla_node_shape() const { return node->xla_shape(); }
+const xla::Shape& Value::xla_shape() const {
+  Node* casted = dynamic_cast<Node*>(node.get());
+  return casted->xla_shape(index);
+}
 
-torch::lazy::hash_t Value::hash() const {
-  return torch::lazy::HashCombine(node->hash(), index);
+const xla::Shape& Value::xla_node_shape() const {
+  Node* casted = dynamic_cast<Node*>(node.get());
+  return casted->xla_shape();
 }
 
+Node* Value::operator->() const { return dynamic_cast<Node*>(node.get()); }
```

Comment:
why is this necessary?

Reply:
This was more of a convenience change because `operator->()` is often used in our code to call the `xla_shape()` from our `Value` like . Adding this will prevent us from manually doing the dynamic cast each time.

But after looking again, I noticed we only do this call couple times in our ops. So I've removed this operator and did manually dynamic casting instead.